pkg/apis/monitoring/v1/podmonitor_types.go

```diff
@@ -31,7 +31,16 @@ const (
 // +k8s:openapi-gen=true
 // +kubebuilder:resource:categories="prometheus-operator",shortName="pmon"
 
-// PodMonitor defines monitoring for a set of pods.
+// The `PodMonitor` custom resource definition (CRD) allows to declaratively define how a dynamic set of pods should be monitored.
+// Which pods are selected to be monitored with the desired configuration is defined using label selections.
+// This allows an organization to introduce conventions around how metrics are exposed, and then following these conventions new pods are automatically discovered, without the need to reconfigure the system.
+//
+// The `PodMonitor` object introduced by the Prometheus Operator discovers pods and generates the relevant configuration for the Prometheus server to monitor them.
```

Comment:
I think we can move it to the dedicated part of the struct, instead of the main section